agent-lib/src/main/java/org/terracotta/angela/agent/AgentController.java

```diff
@@ -229,6 +231,55 @@ public boolean installVoter(InstanceId instanceId, TerracottaVoter terracottaVot
     return true;
   }
 
+  public boolean installClusterTool(InstanceId instanceId, String hostName, Distribution distribution,
+                                    License license, String kitInstallationName, SecurityRootDirectory securityRootDirectory,
+                                    TerracottaCommandLineEnvironment tcEnv) {
+    ToolInstall clusterToolInstall = clusterToolInstalls.get(instanceId);
+    if (clusterToolInstall == null) {
+      RemoteKitManager kitManager = new RemoteKitManager(instanceId, distribution, kitInstallationName);
+      if (!kitManager.isKitAvailable()) {
+        return false;
+      }
+
+      logger.info("Installing kit on host {} from {}", hostName, distribution);
+      File kitDir = kitManager.installKit(license, Collections.singletonList(hostName));
+      File workingDir = kitManager.getWorkingDir().toFile();
+      clusterToolInstalls.computeIfAbsent(instanceId, id -> {
+        Function<String[], ToolExecutionResult> operation = command -> {
+          DistributionController distributionController = distribution.createDistributionController();
+          return distributionController.invokeClusterTool(kitDir, workingDir, securityRootDirectory, tcEnv, command);
```

Comment:
The purpose of this lambda was to avoid duplication of classes, which essentially do the same thing, and differ in only the operation they need to execute.

This lambda is invoked from within `AgentController` only (e.g. `clusterToolInstall.getInstance().execute(command)`).

Reply:
ok I get it